src/Analysis/Ast/Impl/Analyzer/Symbols/FunctionEvaluator.cs

```diff
@@ -38,10 +38,9 @@ internal sealed class FunctionEvaluator : MemberEvaluator {
         public FunctionEvaluator(ExpressionEval eval, PythonFunctionOverload overload)
             : base(eval, overload.FunctionDefinition) {
             _overload = overload;
-            _function = overload.ClassMember ?? throw new NullReferenceException(nameof(overload.ClassMember));
+            _function = overload.ClassMember ?? throw new ArgumentNullException(nameof(overload));
```

Comment:
If the exception is referring to the overload, should the null check be done above? What are these guarding against? I'm also noting that FunctionDefinition gets passed to the base before this code would run, too.